localstack/services/awslambda/lambda_executors.py

```diff
@@ -1192,12 +1192,18 @@ def reset_default_env_variables(env_vars_before):
                 os.environ.pop(env_name, None)
 
     def execute_go_lambda(self, event, context, main_file, func_details=None):
-        event_json_string = "%s" % (json.dumps(event) if event else "{}")
-        cmd = "AWS_LAMBDA_FUNCTION_HANDLER=%s AWS_LAMBDA_EVENT_BODY='%s' %s" % (
-            main_file,
-            event_json_string,
-            GO_LAMBDA_RUNTIME,
-        )
+        cmd = GO_LAMBDA_RUNTIME
+
+        if func_details:
+            func_details.envvars["AWS_LAMBDA_FUNCTION_HANDLER"] = main_file
+            func_details.envvars["AWS_LAMBDA_EVENT_BODY"] = json.dumps(json_safe(event))
+        else:
+            func_details = {}
+            func_details.envvars = {
```

Comment:
Thanks for the fix in this PR @pinzon ! 👍 

Assigning an attribute to a dict will likely not work (raising `AttributeError`). I think we can safely assume that `func_details` should be passed here (`execute_go_lambda(..)` is only called from one place currently). I think we can simply add a `LOG.warning(...)` message in the `else` branch here.. Thanks